### ISSUE TITLE
[FEATURE] Stacked templates

### ISSUE THREAD

**Issue by esmeetewinkel** [IDEMSInternational/open-app-builder#2465]
**What?**
Ability to launch and display stacked templates, with small top offset from parent template. 

![image](https://github.com/user-attachments/assets/1128518c-a27e-40da-9662-6d1deee5725f)
[Figma](https://www.figma.com/design/CYnvCNWHbgMdXlqBeBnc4T/ParentApp-for-Kids?node-id=3-4585&node-type=canvas&t=RewbTDAQNdK92pj6-0)

**Comment by chrismclarke** [comment]
@jfmcquade 
As a first approach would likely recommend:

- [ ] Create `stack_navigation` service that registers action handlers for `nav_stack push` and `nav_stack pop` actions.
- [ ] Add support for pushing new template on top of existing content as part of stack. Support stack multiple levels, layered offset
- [ ] Add support for popping top level from stack

Would also be good whilst working on these to setup a decent testing environment (as more complex behaviour will need to be built on top). Ideally component spec, service spec and debug sheet.

Once an initial PR is merged can move onto more complex tasks

- [ ] Navigation within a stack
- [ ] Handling completed/uncompleted actions triggered within a stack (and how to author from nav_stack push action)
- [ ] Popups within a stack
- [ ] Custom stack styling (e.g. as an alternative to full-screen popups or regular popups)

**Comment by esmeetewinkel** [comment]
@jfmcquade It seems we haven't linked any of the PRs that went in to this issue. Could you add a comment so that they are linked? I think it might make sense to close this issue and add a follow-up issue for any outstanding tasks.

**Comment by jfmcquade** [comment]
Relevant PRs:

#2550 
#2656
#2683 

I'm closing this issue in favour of #2697, which inherits the uncompleted issue points from this one
